QATester/com/generic/QATester/QATester.java

```diff
@@ -0,0 +1,176 @@
+package com.generic.QATester;
+
+import static org.junit.jupiter.api.Assertions.*;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import com.generic.model.Warehouse;
+import com.generic.tracker.WarehouseTracker;
+
+
+import com.generic.model.FreightType;
+import com.generic.model.Shipment;
+
+
+/**
+ * A Junit class for unit testing
+ * @author Seyi Ola
+ *
+ */
+
+public class QATester {
+	
+	
+	Warehouse warehouse1;
+	Warehouse warehouse2;
+	Warehouse warehouse3;
+	
+	WarehouseTracker wTracker;
+	
+	Shipment shipment1;
+	Shipment shipment2;
+	Shipment shipment3;
+	Shipment shipment4;
+	
+	
+	 
+	
+	public QATester() {
+		//Consider using a singleton pattern for WarehouseController Class
+		wTracker = new WarehouseTracker();
+		
+	
+		// first warehouse and shipment
+		wTracker.addWarehouse(new Warehouse(12513, true));
+		wTracker.addShipment(12513, new Shipment("48934j", FreightType.valueOf("air"), 84, 1515354694451L));
+		
+		
+		// second valid warehouse and shipment
+		wTracker.addWarehouse(new Warehouse(15566, true));
+		wTracker.addShipment(15566, new Shipment("1adf4", FreightType.valueOf("truck") , 354, 1515354694451L));
+		
+		//second warehouse and shipment
+		wTracker.addWarehouse(new Warehouse(15566, false));
+		
+		
+		
+		//Initialize a bunch of shipment objects
+		shipment1 = new Shipment("123tr", FreightType.valueOf("rail"), 74.0, 1515354694451L);
+		shipment2 = new Shipment("4231e", FreightType.valueOf("ship"), 88.0, 1515254694451L);
+		
+		
+		assertEquals(1, wTracker.addShipment(15566, shipment2));
+
+		
+	}
+	
+	/**
+	 * Test to ensure that a warehouse key is always
+	 * mapped to object's instance. This implies that
+	 * the warehouse key will always be same as the
+	 * warehouseID field
+	 * @param warehouseID pre-defined(already existing) set of warehouseIDs
+	 */
+	@ParameterizedTest
+	@ValueSource(ints = {12513,15566})
+	void testWarehouseTrackerIDKeyMapping(int warehouseID) 
+	{
+		assertEquals(warehouseID, wTracker.getWarehouse(warehouseID).getWarehouseID());
```

Comment:
Tests should be written in the AAA format-
Arrange, Act, Assert.

Reply:
Noted